This Jupyter notebook will be used to carry out Capstone project for IBM data science professional certification.

Part-1: Scrap Wikipedia page to get Toronto postal code data and transform them into pandas dataframe


In [3]:
import pandas as pd
import numpy as np

In [1]:
print ('Scrapping and getting Toronto postal code data')

Scrapping and getting Toronto postal code data


In [4]:
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
#This returns list of DataFrames 
df_list = pd.read_html(url)
#Total number of tables found
len(df_list)

3

In [12]:
#To access the Toronto postal code table, simply access that corresponding element of the list.
from pandas import DataFrame
df_toronto=df_list[0]
df_toronto

Postal Code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
..          ...               ...   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                         Neighbourhood  
0                                         Not assigned  
1                                         Not assigned  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
..                                                 ...  
175                                       Not assigned  
176                                       Not assigned  
177                                       Not assigned  
178  Mimico NW, The Queensway West, South of Bloor,...  
179                                       Not assigned  

[180 rows x 3 columns]

In [14]:
#Remove all the rows from the dataframe where Borough is 'Not Assigned'

df_toronto.drop(df_toronto[df_toronto.Borough=='Not assigned'].index, inplace=True)


Update the DataFrame - For Neighbourhood with 'Not Assigned' update is made to assign the Borrough name as Neighbourhood name


In [29]:
#Replace all 'Not Assigned' neighbourhood with the corresponding borrough name
#Use Numpy library to crry out the replacement
df_toronto['Neighbourhood']=np.where(df_toronto['Neighbourhood']=='Not assigned', df_toronto['Borough'], df_toronto['Neighbourhood'])
df_toronto    


Postal Code           Borough  \
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
5           M6A        North York   
6           M7A  Downtown Toronto   
..          ...               ...   
160         M8X         Etobicoke   
165         M4Y  Downtown Toronto   
168         M7Y      East Toronto   
169         M8Y         Etobicoke   
178         M8Z         Etobicoke   

                                         Neighbourhood  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
5                     Lawrence Manor, Lawrence Heights  
6          Queen's Park, Ontario Provincial Government  
..                                                 ...  
160      The Kingsway, Montgomery Road, Old Mill North  
165                               Church and Wellesley  
168  Business reply mail Processing Centre, South C...  
169  Old Mill South, King's Mill Park, Sunnylea, Hu...  
178  Mimico NW, The Queensway West, South of Bloor,...  

[103 rows x 3 columns]

In [20]:
#Print Number of rows of the final dataframe
df_toronto.shape

(103, 3)

**********************This is the end of Part-1 of the assignment******************************

Part-2: Retrieving geographical co-ordinates of the relevant 103 postal codes available in the df_toronto dataframe and add latitude and longitude column in the dataframe

Here I have used the CSV file in the location http://cocl.us/Geospatial_data to get the matching co-ordinates for relevant Postal codes

In [22]:
#Read the CSV file into a dataframe
!wget -O Geospatial_Coordinates.csv http://cocl.us/Geospatial_data/Geospatial_Coordinates.csv

coordinate_df = pd.read_csv("Geospatial_Coordinates.csv")

--2021-01-28 02:09:51--  http://cocl.us/Geospatial_data/Geospatial_Coordinates.csv
Resolving cocl.us (cocl.us)... 169.63.96.176, 169.63.96.194
Connecting to cocl.us (cocl.us)|169.63.96.176|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://cocl.us/Geospatial_data/Geospatial_Coordinates.csv [following]
--2021-01-28 02:09:51--  https://cocl.us/Geospatial_data/Geospatial_Coordinates.csv
Connecting to cocl.us (cocl.us)|169.63.96.176|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://ibm.box.com/shared/static/9afzr83pps4pwf2smjjcf1y5mvgb18rr.csv [following]
--2021-01-28 02:09:52--  https://ibm.box.com/shared/static/9afzr83pps4pwf2smjjcf1y5mvgb18rr.csv
Resolving ibm.box.com (ibm.box.com)... 185.235.236.197
Connecting to ibm.box.com (ibm.box.com)|185.235.236.197|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: /public/static/9afzr83pps4pwf2smjjcf1y5mvgb18rr.

In [23]:
coordinate_df

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
..          ...        ...        ...
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

[103 rows x 3 columns]

In [31]:
#Add Latitude and Longitude columns in the df_toronto dataframe
toronto_postalcodes_coordinates_df=pd.merge(df_toronto, coordinate_df, how='inner', on=['Postal Code'])



In [32]:
#Print the consolidated dataframe
toronto_postalcodes_coordinates_df

Postal Code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                         Neighbourhood   Latitude  Longitude  
0                                            Parkwoods  43.753259 -79.329656  
1                                     Victoria Village  43.725882 -79.315572  
2                            Regent Park, Harbourfront  43.654260 -79.360636  
3                     Lawrence Manor, Lawrence Heights  43.718518 -79.464763  
4          Queen's Park, Ontario Provincial Government  43.662301 -79.389494  
..                                                 ...        ...        ...  
98       The Kingsway, Montgomery Road, Old Mill North  43.653654 -79.506944  
99                                Church and Wellesley  43.665860 -79.383160  
100  Business reply mail Processing Centre, South C...  43.662744 -79.321558  
101  Old Mill South, King's Mill Park, Sunnylea, Hu...  43.636258 -79.498509  
102  Mimico NW, The Queensway West, South of Bloor,...  43.628841 -79.520999  

[103 rows x 5 columns]

********** This Ends Part-2 of the assignments ************

Part-3: Explore and cluster the neighbourhood of Toronto

Boroughs will be clustered. Consider all boroughs present in the above dataframe where the word 'Toronto'present in the borough name.

In [34]:
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
#Use geopy library to get the latitude and longitude values of Toronto.
address = 'Toronto, ON'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Toronto are {}, {}.'.format(latitude, longitude))


The geograpical coordinate of Toronto are 43.6534817, -79.3839347.


In [35]:
#Create neighbourhood dataframe considering boroughs having toronto word in it
neighbour_df=toronto_postalcodes_coordinates_df[toronto_postalcodes_coordinates_df.Borough.str.contains('Toronto',case=False)]

In [36]:
#Print the dataframe
neighbour_df

Postal Code           Borough  \
2           M5A  Downtown Toronto   
4           M7A  Downtown Toronto   
9           M5B  Downtown Toronto   
15          M5C  Downtown Toronto   
19          M4E      East Toronto   
20          M5E  Downtown Toronto   
24          M5G  Downtown Toronto   
25          M6G  Downtown Toronto   
30          M5H  Downtown Toronto   
31          M6H      West Toronto   
36          M5J  Downtown Toronto   
37          M6J      West Toronto   
41          M4K      East Toronto   
42          M5K  Downtown Toronto   
43          M6K      West Toronto   
47          M4L      East Toronto   
48          M5L  Downtown Toronto   
54          M4M      East Toronto   
61          M4N   Central Toronto   
62          M5N   Central Toronto   
67          M4P   Central Toronto   
68          M5P   Central Toronto   
69          M6P      West Toronto   
73          M4R   Central Toronto   
74          M5R   Central Toronto   
75          M6R      West Toronto   
79          M4S   Central Toronto   
80          M5S  Downtown Toronto   
81          M6S      West Toronto   
83          M4T   Central Toronto   
84          M5T  Downtown Toronto   
86          M4V   Central Toronto   
87          M5V  Downtown Toronto   
91          M4W  Downtown Toronto   
92          M5W  Downtown Toronto   
96          M4X  Downtown Toronto   
97          M5X  Downtown Toronto   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   

                                         Neighbourhood   Latitude  Longitude  
2                            Regent Park, Harbourfront  43.654260 -79.360636  
4          Queen's Park, Ontario Provincial Government  43.662301 -79.389494  
9                             Garden District, Ryerson  43.657162 -79.378937  
15                                      St. James Town  43.651494 -79.375418  
19                                         The Beaches  43.676357 -79.293031  
20                                         Berczy Park  43.644771 -79.373306  
24                                  Central Bay Street  43.657952 -79.387383  
25                                            Christie  43.669542 -79.422564  
30                            Richmond, Adelaide, King  43.650571 -79.384568  
31                        Dufferin, Dovercourt Village  43.669005 -79.442259  
36   Harbourfront East, Union Station, Toronto Islands  43.640816 -79.381752  
37                            Little Portugal, Trinity  43.647927 -79.419750  
41                        The Danforth West, Riverdale  43.679557 -79.352188  
42            Toronto Dominion Centre, Design Exchange  43.647177 -79.381576  
43        Brockton, Parkdale Village, Exhibition Place  43.636847 -79.428191  
47                      India Bazaar, The Beaches West  43.668999 -79.315572  
48                      Commerce Court, Victoria Hotel  43.648198 -79.379817  
54                                     Studio District  43.659526 -79.340923  
61                                       Lawrence Park  43.728020 -79.388790  
62                                            Roselawn  43.711695 -79.416936  
67                                    Davisville North  43.712751 -79.390197  
68     Forest Hill North & West, Forest Hill Road Park  43.696948 -79.411307  
69                       High Park, The Junction South  43.661608 -79.464763  
73                   North Toronto West, Lawrence Park  43.715383 -79.405678  
74                 The Annex, North Midtown, Yorkville  43.672710 -79.405678  
75                              Parkdale, Roncesvalles  43.648960 -79.456325  
79                                          Davisville  43.704324 -79.388790  
80                      University of Toronto, Harbord  43.662696 -79.400049  
81                                  Runnymede, Swansea  43.651571 -79.484450  
83                         Moore Park, Summerhill East  43.689574 -79.383160  
84           Kensington Market, Chinatown, Grange Park  43.653206 -79.4000

In [38]:
#Check the number of records - this corresponds to the total number of neighbourhood boroughs of Toronto
neighbour_df.shape

(39, 5)

In [52]:
#Import folium library
!pip install folium
print('Libraries imported successfully.')

     |████████████████████████████████| 94 kB 6.6 MB/s  eta 0:00:01
Libraries imported successfully.


In [55]:
import folium # map rendering library
# create map of Toronto using latitude and longitude values
map_toronto = folium.Map(location=[latitude, longitude], zoom_start=11)

#Boroughs contains 'Toronto' are considered as neighbourhood of Toronto
# add markers to map
for lat, lng, label in zip(neighbour_df['Latitude'], neighbour_df['Longitude'], neighbour_df['Borough']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='red',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_toronto)  
    
map_toronto

Above is the map of the City of Toronto with the neighbourhood boroughs considered.
As the next step, we will use Foursquare to get the neighbourhood data and do clustering

In [56]:
#Define Foursquare Credentials and Version
CLIENT_ID = 'LMAMZ1D4VRD5D5IYI11DU4ZDXYQW1TAJKV0IR5EIA0SDKSRU' # your Foursquare ID
CLIENT_SECRET = 'R2KDSHOH0JLMIQLOMEO35JHFXVOBLFPDBFKRMGJDPUYWE1EB' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version
LIMIT = 100 # A default Foursquare API limit value

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: LMAMZ1D4VRD5D5IYI11DU4ZDXYQW1TAJKV0IR5EIA0SDKSRU
CLIENT_SECRET:R2KDSHOH0JLMIQLOMEO35JHFXVOBLFPDBFKRMGJDPUYWE1EB


In [57]:
import json # library to handle JSON files
import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

There are four different unique borough names are there - Downtown Toronto, Central Toronto, West Toronto & East Toronto
As part of the clustering exercise, 20 venues within the radious of 500 meter of each neighbourhood data is being fetched.


In [76]:
LIMIT = 20 # limit of number of venues returned by Foursquare API
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Borough', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

For Each of the Neighbourhoods, get the venue details and create a dataframe.

In [77]:
toronto_venues_df = getNearbyVenues(names=neighbour_df['Borough'],
                                   latitudes=neighbour_df['Latitude'],
                                   longitudes=neighbour_df['Longitude']
                                  )

Downtown Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
East Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
West Toronto
Downtown Toronto
West Toronto
East Toronto
Downtown Toronto
West Toronto
East Toronto
Downtown Toronto
East Toronto
Central Toronto
Central Toronto
Central Toronto
Central Toronto
West Toronto
Central Toronto
Central Toronto
West Toronto
Central Toronto
Downtown Toronto
West Toronto
Central Toronto
Downtown Toronto
Central Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
Downtown Toronto
East Toronto


In [78]:
#View the size of the dataframe
print(toronto_venues_df.shape)
toronto_venues_df.head()

(641, 7)


Borough  Neighborhood Latitude  Neighborhood Longitude  \
0  Downtown Toronto               43.65426              -79.360636   
1  Downtown Toronto               43.65426              -79.360636   
2  Downtown Toronto               43.65426              -79.360636   
3  Downtown Toronto               43.65426              -79.360636   
4  Downtown Toronto               43.65426              -79.360636   

                    Venue  Venue Latitude  Venue Longitude  \
0        Roselle Desserts       43.653447       -79.362017   
1           Tandem Coffee       43.653559       -79.361809   
2  Cooper Koo Family YMCA       43.653249       -79.358008   
3     Body Blitz Spa East       43.654735       -79.359874   
4          Impact Kitchen       43.656369       -79.356980   

        Venue Category  
0               Bakery  
1          Coffee Shop  
2  Distribution Center  
3                  Spa  
4           Restaurant

In [79]:
#Check how many venues are returned per neighbourhood
toronto_venues_df.groupby('Borough').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Borough                                                                  
Central Toronto                      90                      90     90   
Downtown Toronto                    358                     358    358   
East Toronto                         82                      82     82   
West Toronto                        111                     111    111   

                  Venue Latitude  Venue Longitude  Venue Category  
Borough                                                            
Central Toronto               90               90              90  
Downtown Toronto             358              358             358  
East Toronto                  82               82              82  
West Toronto                 111              111             111

In [80]:
print('There are {} uniques categories.'.format(len(toronto_venues_df['Venue Category'].unique())))

There are 173 uniques categories.


Now analysis of each unique Neighbourhood will be done - Here we have four unique categories.

In [81]:
# one hot encoding
toronto_onehot = pd.get_dummies(toronto_venues_df[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
toronto_onehot['Borough'] = toronto_venues_df['Borough'] 

# move neighborhood column to the first column
fixed_columns = [toronto_onehot.columns[-1]] + list(toronto_onehot.columns[:-1])
toronto_onehot = toronto_onehot[fixed_columns]

toronto_onehot.head()

Borough  Airport  Airport Food Court  Airport Gate  \
0  Downtown Toronto        0                   0             0   
1  Downtown Toronto        0                   0             0   
2  Downtown Toronto        0                   0             0   
3  Downtown Toronto        0                   0             0   
4  Downtown Toronto        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Antique Shop  Art Gallery  ...  Theater  Theme Restaurant  \
0             0            0  ...        0                 0   
1             0            0  ...        0                 0   
2             0            0  ...        0                 0   
3             0            0  ...        0                 0   
4             0            0  ...        0                 0   

   Toy / Game Store  Trail  Train Station  Vegetarian / Vegan Restaurant  \
0                 0      0              0                              0   
1                 0      0              0                              0   
2                 0      0              0                              0   
3                 0      0              0                              0   
4                 0      0              0                              0   

   Vietnamese Restaurant  Wine Bar  Wine Shop  Yoga Studio  
0                      0         0          0            0  
1                      0         0          0            0  
2                      0         0          0            0  
3                      0         0          0            0  
4                      0         0          0            0  

[5 rows x 174 columns]

let's group rows by neighborhood and by taking the mean of the frequency of occurrence of each category

In [82]:
toronto_grouped = toronto_onehot.groupby('Borough').mean().reset_index()
toronto_grouped

Borough   Airport  Airport Food Court  Airport Gate  \
0   Central Toronto  0.000000            0.000000      0.000000   
1  Downtown Toronto  0.002793            0.002793      0.002793   
2      East Toronto  0.000000            0.000000      0.000000   
3      West Toronto  0.000000            0.000000      0.000000   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0        0.000000          0.00000          0.000000             0.011111   
1        0.005587          0.00838          0.005587             0.008380   
2        0.000000          0.00000          0.000000             0.012195   
3        0.000000          0.00000          0.000000             0.000000   

   Antique Shop  Art Gallery  ...   Theater  Theme Restaurant  \
0      0.000000     0.000000  ...  0.000000          0.000000   
1      0.000000     0.011173  ...  0.011173          0.002793   
2      0.000000     0.000000  ...  0.000000          0.000000   
3      0.009009     0.009009  ...  0.009009          0.000000   

   Toy / Game Store     Trail  Train Station  Vegetarian / Vegan Restaurant  \
0          0.011111  0.022222       0.000000                       0.000000   
1          0.000000  0.002793       0.002793                       0.011173   
2          0.000000  0.024390       0.000000                       0.000000   
3          0.000000  0.000000       0.000000                       0.009009   

   Vietnamese Restaurant  Wine Bar  Wine Shop  Yoga Studio  
0               0.011111  0.000000   0.000000     0.011111  
1               0.002793  0.002793   0.000000     0.005587  
2               0.000000  0.000000   0.000000     0.024390  
3               0.009009  0.009009   0.009009     0.009009  

[4 rows x 174 columns]

Run k-means to cluster the neighborhood (Borough) into 4 clusters

In [88]:
# import k-means from clustering stage
from sklearn.cluster import KMeans

# set number of clusters
kclusters = 4

#Drop the 'Borough' column as the same contains String variables - Index 1 signifies column drop
toronto_grouped_clustering = toronto_grouped.drop('Borough', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(toronto_grouped_clustering)

# check cluster labels generated for each row in the dataframe
labels_= kmeans.labels_

One new dataframe will be created including clusters and top 10 venues of each neighbour (or Borough in current case)

In [89]:
toronto_grouped_clustering["Labels"] = labels_

In [92]:
toronto_grouped_clustering

Airport  Airport Food Court  Airport Gate  Airport Lounge  \
0  0.000000            0.000000      0.000000        0.000000   
1  0.002793            0.002793      0.002793        0.005587   
2  0.000000            0.000000      0.000000        0.000000   
3  0.000000            0.000000      0.000000        0.000000   

   Airport Service  Airport Terminal  American Restaurant  Antique Shop  \
0          0.00000          0.000000             0.011111      0.000000   
1          0.00838          0.005587             0.008380      0.000000   
2          0.00000          0.000000             0.012195      0.000000   
3          0.00000          0.000000             0.000000      0.009009   

   Art Gallery  Art Museum  ...  Theme Restaurant  Toy / Game Store     Trail  \
0     0.000000    0.000000  ...          0.000000          0.011111  0.022222   
1     0.011173    0.002793  ...          0.002793          0.000000  0.002793   
2     0.000000    0.000000  ...          0.000000          0.000000  0.024390   
3     0.009009    0.000000  ...          0.000000          0.000000  0.000000   

   Train Station  Vegetarian / Vegan Restaurant  Vietnamese Restaurant  \
0       0.000000                       0.000000               0.011111   
1       0.002793                       0.011173               0.002793   
2       0.000000                       0.000000               0.000000   
3       0.000000                       0.009009               0.009009   

   Wine Bar  Wine Shop  Yoga Studio  Labels  
0  0.000000   0.000000     0.011111       0  
1  0.002793   0.000000     0.005587       2  
2  0.000000   0.000000     0.024390       1  
3  0.009009   0.009009     0.009009       3  

[4 rows x 174 columns]

In [91]:

#We can easily check the centroid values by averaging the features in each cluster.
toronto_grouped_clustering.groupby('Labels').mean()

Airport  Airport Food Court  Airport Gate  Airport Lounge  \
Labels                                                               
0       0.000000            0.000000      0.000000        0.000000   
1       0.000000            0.000000      0.000000        0.000000   
2       0.002793            0.002793      0.002793        0.005587   
3       0.000000            0.000000      0.000000        0.000000   

        Airport Service  Airport Terminal  American Restaurant  Antique Shop  \
Labels                                                                         
0               0.00000          0.000000             0.011111      0.000000   
1               0.00000          0.000000             0.012195      0.000000   
2               0.00838          0.005587             0.008380      0.000000   
3               0.00000          0.000000             0.000000      0.009009   

        Art Gallery  Art Museum  ...   Theater  Theme Restaurant  \
Labels                           ...                               
0          0.000000    0.000000  ...  0.000000          0.000000   
1          0.000000    0.000000  ...  0.000000          0.000000   
2          0.011173    0.002793  ...  0.011173          0.002793   
3          0.009009    0.000000  ...  0.009009          0.000000   

        Toy / Game Store     Trail  Train Station  \
Labels                                              
0               0.011111  0.022222       0.000000   
1               0.000000  0.024390       0.000000   
2               0.000000  0.002793       0.002793   
3               0.000000  0.000000       0.000000   

        Vegetarian / Vegan Restaurant  Vietnamese Restaurant  Wine Bar  \
Labels                                                                   
0                            0.000000               0.011111  0.000000   
1                            0.000000               0.000000  0.000000   
2                            0.011173               0.002793  0.002793   
3                            0.009009               0.009009  0.009009   

        Wine Shop  Yoga Studio  
Labels                          
0        0.000000     0.011111  
1        0.000000     0.024390  
2        0.000000     0.005587  
3        0.009009     0.009009  

[4 rows x 173 columns]

*********This End the Part-3 of the assignment************